src/stdlib_linalg_kronecker.fypp

```diff
@@ -0,0 +1,28 @@
+#:include "common.fypp"
+#:set RCI_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES + INT_KINDS_TYPES
+submodule (stdlib_linalg) stdlib_linalg_kronecker
+
+  implicit none
+
+contains
+
+  #:for k1, t1 in RCI_KINDS_TYPES
+    pure module function kronecker_product_${t1[0]}$${k1}$(A, B) result(C)
+      ${t1}$, intent(in) :: A(:,:), B(:,:)
+      ${t1}$ :: C(size(A,dim=1)*size(B,dim=1),size(A,dim=2)*size(B,dim=2))
+      integer :: m1, n1, m2, n2, maxM1, maxN1, maxM2, maxN2
+      
+      maxM1 = size(A, dim=1)
+      maxN1 = size(A, dim=2)
+      maxM2 = size(B, dim=1)
+      maxN2 = size(B, dim=2)
+      
+      do n1=1, maxN1
+         do m1=1, maxM1
+            ! We use the numpy.kron convention for ordering of the matrix elements
```

Comment:
I suggest that you refer to a reference that does not depend on a language. For example, you could refer to Wikipedia (that follows the same ordering).

Reply:
Sure, I'll do that.